src/pythonegardia/egardiaserver.py

```diff
@@ -1,152 +1,115 @@
 import argparse
-import sys
 import logging
-import datetime
-import homeassistant.remote as remote
-import time
+import socketserver
+import sys
+import threading
 
 _LOGGER = logging.getLogger(__name__)
 
+DEFAULT_RESPONSE = "HTTP/1.1 200 OK\n\nHello, World!"
+
+
+class EgardiaServer(socketserver.TCPServer, threading.Thread):
+    """Egardia server passing along alarms."""
+
+    def __init__(self, host, port):
+        """Prepare server."""
+        threading.Thread.__init__(self)
+        self._address = (host, port)
+        self.callbacks = []
+        socketserver.TCPServer.__init__(
+            self, self._address, EgardiaServerHandler, bind_and_activate=False)
+        self.status = ''
 
-class EgardiaServer(object):
-    """
-    Egardia server passing along alarms
-    """
-    def __init__(self, port, hasshost, password, haport, ssl, retrycount, waittime):
-        self._port = port
-        self._hasshost = hasshost
-        self._password = password
-        self._haport = haport
-        self._ssl = ssl
-        self._retrycount = retrycount
-        self._waittime = waittime
-        msg = "Sleeping to give Home Assistant a moment to start"
-        _LOGGER.info(msg)
-        print(msg)
-        time.sleep(waittime)
+    def bind(self):
+        """Bind and listen to address."""
         try:
-            validationresult = "init"
-            numretries = 0
-            while validationresult != "ok" and numretries <= self._retrycount:
-                self._api = remote.API(self._hasshost, self._password,
-                                       self._haport,self._ssl)
-                validationresult = str(remote.validate_api(self._api))
-                numretries = numretries+1
-                if validationresult != "ok":
-                   msg = "Connecting to Home Assistant failed, "
-                   msg = msg + "retrying..."
-                   _LOGGER.info(msg)
-                   print(msg)
-                   time.sleep(5)
-            if validationresult != "ok":
-               msg = "Error connection to Home Assistant: "
-               msg = msg + validationresult
-               _LOGGER.error(msg)
-               print(msg)
-               sys.exit(2)
-            else:
-               msg = "Succesfully connected to Home Assistant"
-               print(msg)
-               _LOGGER.info(msg)
-        except:
-            e = sys.exc_info()[0]
-            msg = "Error connecting to Home Assistant: "
-            msg = msg + str(e)+", please check settings."
-            print(msg)
-            _LOGGER.error(msg)
-            sys.exit(2)
-        listen_socket = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
-        listen_socket.setsockopt(socket.SOL_SOCKET, socket.SO_REUSEADDR, 1)
+            self.allow_reuse_address = True
+            self.server_bind()
+            self.server_activate()
+        except OSError:
+            self.server_close()
+            _LOGGER.exception(
```

Comment:
This should probably be set to error level before merging this. I changed it to exception level when debugging.